resources/indexer.py

```diff
@@ -18,290 +19,292 @@
+    def name_to_year(name):
+        """Return the year of the season start for the season's name."""
+        if name:
+            for season_year, season_name in IceCubeSeason.SEASONS.items():
+                if season_name == name:
+                    return int(season_year)
+            raise Exception(f"No season found for {name}.")
+        else:
+            return None
+
+    @staticmethod
+    def year_to_name(season_year):
+        """Return the season's name for the year of the season start."""
+        if season_year:
+            try:
+                return IceCubeSeason.SEASONS[str(season_year)]
+            except KeyError:
+                raise Exception(f"No season found for {season_year}.")
+        else:
+            return None
+
+
+class ProcessingLevel:
+    """Wrapper static class encapsulating processing-level parsing logic and levels constants."""
+    PFRaw = "PFRaw"
+    PFFilt = "PFFilt"
+    PFDsT = "PFDsT"
+    L2 = "L2"
+    L3 = "L3"
+    L4 = "L4"
+
+    @staticmethod
+    def from_path(dir_path):
+        """Return the processing level parsed from the dir path, case insensitive."""
+        # Ex: /data/exp/IceCube/2013/filtered/level2/1229/Run00123579_0/
+        search_strings = {
+            "PFRaw": ProcessingLevel.PFRaw,
+            "PFFilt": ProcessingLevel.PFFilt,
+            "PFDsT": ProcessingLevel.PFDsT,
+            "Level2": ProcessingLevel.L2,
+            "Level3": ProcessingLevel.L3,
+            "Level4": ProcessingLevel.L4
+        }
+        dir_path_upper = dir_path.upper()
+        for string, level in search_strings.items():
+            if string.upper() in dir_path_upper:
+                return level
+        raise Exception(f"Cannot detect processing level, {dir_path}.")
+
+
+class BasicFileMetadata:
+    """Metadata for basic files"""
+    file = None
+    site = ""
+
+    def __init__(self, file, site):
+        self.file = file
+        self.site = site
+
+    def generate(self):
+        """Gather the file's metadata."""
+        metadata = dict()
+        metadata['logical_name'] = self.file.path
+        metadata['checksum'] = {'sha512': self.sha512sum()}
+        metadata['file_size'] = self.file.stat().st_size
+        metadata['locations'] = self._get_locations()
+        metadata['create_date'] = date.fromtimestamp(os.path.getctime(self.file.path)).isoformat()
+        return metadata
+
+    def sha512sum(self):
+        """Return the SHA512 checksum of the file given by path."""
+        bufsize = 4194304
+        sha = hashlib.new('sha512')
+        with open(self.file.path, 'rb', buffering=0) as file:
+            line = file.read(bufsize)
+            while line:
+                sha.update(line)
+                line = file.read(bufsize)
+        return sha.hexdigest()
+
+    def _get_locations(self):
+        """Return the locations object."""
+        if '.tar' in self.file.name:
+            return [{'site': self.site, 'path': self.file.path, 'archive': True}]
```

Comment:
This is not what the archive flag is for.  archive means things like the NERSC or DESY archive, which are not directly accessible.